stdlib/public/Darwin/Foundation/String.swift

```diff
@@ -22,10 +22,61 @@
 
 extension String {
   public init(_ cocoaString: NSString) {
-    self = String(_cocoaString: cocoaString)
+    self = String._unconditionallyBridgeFromObjectiveC(cocoaString)
   }
 }
 
+@_effects(readonly)
+private func _getClass(_ obj: NSString) -> AnyClass {
+  return object_getClass(obj)!
+}
+
+@_effects(readonly)
+private func _length(_ obj: NSString) -> Int {
+  return CFStringGetLength(obj as CFString)
+}
+
+@_effects(readonly)
+private func _copyString(_ obj: NSString) -> NSString {
+  return CFStringCreateCopy(kCFAllocatorSystemDefault, obj as CFString)
+}
+
+@_effects(releasenone)
+private func _getBytes(_ theString: NSString,
+                       _ range: CFRange,
+                       _ encoding: CFStringEncoding,
+                       _ buffer: UnsafeMutablePointer<UInt8>!,
+                       _ maxBufLen: CFIndex,
+                       _ usedBufLen: UnsafeMutablePointer<CFIndex>!
+) -> CFIndex {
+  return CFStringGetBytes(
+    theString as CFString,
+    range,
+    CFStringBuiltInEncodings.UTF8.rawValue,
+    0, //lossByte,
+    false, //isExternalRepresentation
+    buffer,
+    maxBufLen,
+    usedBufLen
+  )
+}
+
+private let (nscfClass, nscfConstantClass): (AnyClass, AnyClass) =
+  (objc_lookUpClass("__NSCFString")!,
+   objc_lookUpClass("__NSCFConstantString")!)
+
+// This works around an ARC optimizer issue that inserts a retain-release pair
+// if you do the obvious thing and just use a static let constant
+@_transparent
+private func _estimatedSmallStringCutoff() -> Int {
+  #if arch(i386) || arch(arm)
+  return 10
+  #else
+  return 15
+  #endif
+}
```

Comment:
For constants like this, we use always-inline computed `var`s instead of `let`s anyways. Any need for this to be transparent instead of always-inline? Transparent has ABI impact IIUC.

Reply:
It does not.

Reply:
docs/TransparentAttr.rst specifically says that it does not imply `@usableFromInline`.

Reply:
Either way, this shouldn't be transparent, especially in the Foundation module.

Reply:
always-inline does not work for this purpose, due to the mentioned optimizer bug

Reply:
Could you link the SR then in the comment?